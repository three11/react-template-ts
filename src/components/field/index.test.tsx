import * as React from 'react';
import { render } from '@testing-library/react';

import { Field } from '.';

test('Field component should render successfully', () => {
	const { asFragment } = render(
		<Field
			register={(): jest.Mock => jest.fn()}
			type="email"
			name="email"
			label="Email Address"
			error={{}}
			placeholder="someone@example.com"
		/>
	);

	expect(asFragment()).toMatchSnapshot();
});
