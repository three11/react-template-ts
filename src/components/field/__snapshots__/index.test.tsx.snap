// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field component should render successfully 1`] = `
<DocumentFragment>
  <div
    class="c-form__field c-form__field--email c-form__field--error"
  >
    <header
      class="c-form__field-head"
    >
      <label
        for="email"
      >
        Email Address
      </label>
      <p
        class="c-form__error"
      />
    </header>
    <input
      id="email"
      name="email"
      placeholder="someone@example.com"
      type="email"
    />
  </div>
</DocumentFragment>
`;
